src/instrumenter/annotations.ts

```diff
@@ -267,242 +316,255 @@ type RawMetaData = {
+                `The "${annotation.type}" annotation is not applicable inside functions`,
+                annotation.original,
+                annotation.annotationFileRange,
+                target
+            );
         }
-    }
 
-    private findAnnotations(
-        raw: StructuredDocumentation,
-        target: AnnotationTarget,
-        source: string,
-        filters: AnnotationFilterOptions
-    ): AnnotationMetaData[] {
-        const rxType = filters.type === undefined ? undefined : new RegExp(filters.type);
-        const rxMsg = filters.message === undefined ? undefined : new RegExp(filters.message);
+        if (target instanceof TryCatchClause) {
+            throw new UnsupportedByTargetError(
+                `The "${annotation.type}" annotation is not applicable to try-catch clauses`,
+                annotation.original,
+                annotation.annotationFileRange,
+                target
+            );
+        }
+    } else {
+        if (
+            annotation.type !== AnnotationType.IfUpdated &&
+            annotation.type !== AnnotationType.IfAssigned &&
+            /**
+             * @todo Figure out handling the difference between syntax sugar macros for state vars
+             * and contract level macros. This is not yet clear.
+             */
+            annotation.type !== AnnotationType.Macro
+        ) {
+            throw new UnsupportedByTargetError(
+                `The "${annotation.type}" annotation is not applicable to state variables`,
+                annotation.original,
+                annotation.annotationFileRange,
+                target
+            );
+        }
 
-        const sourceInfo = raw.sourceInfo;
+        if (!(target.vScope instanceof ContractDefinition)) {
+            throw new UnsupportedByTargetError(
+                `The "${annotation.type}" annotation is only applicable to state variables`,
+                annotation.original,
+                annotation.annotationFileRange,
+                target
+            );
+        }
+    }
+}
 
-        const meta: RawMetaData = {
-            target: target,
-            node: raw,
-            text: raw.extractSourceFragment(source),
-            loc: [sourceInfo.offset, sourceInfo.length]
-        };
+function findAnnotations(
+    raw: StructuredDocumentation,
+    target: AnnotationTarget,
+    source: string,
+    compilerVersion: string,
+    filters: AnnotationFilterOptions,
+    macros: Map<string, MacroDefinition>
+): AnnotationMetaData[] {
+    const rxType = filters.type === undefined ? undefined : new RegExp(filters.type);
+    const rxMsg = filters.message === undefined ? undefined : new RegExp(filters.message);
 
-        const result: AnnotationMetaData[] = [];
+    const sourceInfo = raw.sourceInfo;
 
-        const rx =
-            /\s*(\*|\/\/\/)\s*#?(if_succeeds|if_updated|if_assigned|invariant|assert|try|require|define\s*[a-zA-Z0-9_]*\s*\([^)]*\))/g;
```

Comment:
Now that '#' is required, I think the `#` in this regex be required?

Reply:
`#` is no longer required due to https://github.com/ConsenSys/scribble/pull/96/commits/e57764f3acdbedbb98e4e3d041446f2210f42531 introduces `macro` keyword/